webapp/src/auth/auth.controller.ts

```diff
@@ -2,66 +2,87 @@ import {
   Body,
   Controller,
   Get,
+  Headers,
   Post,
-  Redirect,
-  Render,
   Request,
-  Session,
+  Res,
   UseGuards,
 } from "@nestjs/common";
 import { AuthService } from "./auth.service";
 import { UsersService } from "../users/users.service";
-import { ApiExcludeController } from "@nestjs/swagger";
+import { ApiCreatedResponse } from "@nestjs/swagger";
+import { LocalAuthGuard } from "./local-auth.guard";
+import { CreateUserDto } from "src/users/dto/create-user.dto";
+import * as argon2 from "argon2";
+import { Response } from "express";
 import { JwtAuthGuard } from "./jwt-auth.guard";
 
 @Controller()
-@ApiExcludeController()
 export class AuthController {
   constructor(
     private authService: AuthService,
     private userService: UsersService,
   ) {}
 
-  @Get("/")
-  @Render("userForm")
-  registerView() {
-    return { action: "/", title: "Register" };
+  @UseGuards(LocalAuthGuard)
+  @Post("login")
+  @ApiCreatedResponse({
+    description: "If credentials are ok, get a JWT",
+  })
+  async signIn(
+    @Body() signInDto: CreateUserDto,
+    @Headers() headers,
+    @Res() response: Response,
+  ) {
+    const token = await this.authService.login(signInDto.username);
+
+    if (headers?.accept && headers.accept === "application/json") {
```

Comment:
You don't need the first check here, do you? If headers.accept is undefined, it won't equals "application/json"